### ISSUE TITLE
Stopping a Routine breaks audio

### ISSUE THREAD

**Issue by Vestal-1** [supercollider/supercollider#6397]
## Environment

* SuperCollider version: 3.13.0
* Jack version: 1.9.22
* Operating system: Void Linux
* Architecture: i686

## Description of the problem
Stopping a playing Routine breaks the audio; in the example, evaluating `Synth(\simpleTest)` or re-evaluating the Routine with `r.play` has no audible output, even though they figure as functional nodes. The scsynth and jack audio servers can't be stopped either, and the postWindow shows this error message:

```supercollider
'/quit' message sent to server 'localhost'.
-> localhost
WARNING: Server 'localhost' failed to quit after 3.0 seconds.
```
In this scenario, the Jack server must be force-stopped externally. If Jack and scsynth are reinitialized, the problem persists (aka, no audio even after the server loads successfully). The only way of getting audio back in track is reinitializing SuperCollider completely; then, the problem can be reproduced. This doesn't seem to be a Jack problem since other applications (such as MPV) can still play sound through the server.

No other error messages are seen in the postWindow.
## Steps to reproduce (+ example)
This problem started occuring after I caused an overload (yikes!) as I **forgot a** `.wait` **in my loop**. I recently reinstalled my system though, but the problem persists! (so maybe just a coincidence?) :(

A simple example I made:
```supercollider
(
SynthDef(\simpleTest, {
    Out.ar(0, SinOsc.ar(440, 0, 0.1)!2) * Env.perc(0.001, 0.1).ar(Done.freeSelf);
}).add;
)

Synth(\simpleTest);

(
var map;
map = Dictionary[
  $s -> \simpleTest
];

r = Routine({
  loop {
    "s.s".do { |character|
      if(character == $.) {
        0.2.wait;
        } {
          if(map[character].notNil) {
            s.bind { Synth(map[character]); 
            0.2.wait;
          };
        };
      };
    };
  };
});
)

r.play;
r.stop;
```
The SynthDef and the Routine must be defined first. Then, evaluate `r.play` to hear it play. In my case, stopping the Routine with `r.stop` breaks the audio and no sound can be produced afterwards. `Synth(\simpleTest)` is included for testing.

Here is a video showing the issue (sc breaks; mpv can still use jack server):

https://github.com/user-attachments/assets/439f1653-2619-4786-bab0-e8b1337cdbfd

PD: In the video, scnvim is used, but **the problem behaves the same way in SuperCollider's IDE**.

## Things I tried (and didn't work):
* Reinstalling the supercollider/jack packages.
* Reinitializing the sclang/jack server.
* Rebooting.
* Testing in another user account.
* Reinstalling absolutely everything :<

This wasn't a problem a couple of months ago (before my incident), so I wonder if it's a bug or if I caused a major issue in my computer after the overload. I'd deeply appreciate any ideas of how to proceed!

**Comment by JordanHendersonMusic** [comment]
Could you try removing the `0.2.wait` from inside the `s.bind`?

**Comment by Vestal-1** [comment]
Hello, thank you for your response. This is embarrasing!! Thanks to your comment, I realized that `s.bind` should be closed before the `0.2.wait`, as in `s.bind{ Synth(map[character]); };`. This solves the issue. The bind is thought to sequence other synths which are also mapped. 

I don't know why SuperCollider behaves the way it does with the 0.2.wait outiside, though, triggering the problem, but I'll close the issue with this comment.

**Comment by JordanHendersonMusic** [comment]
No worries! 

You also need to call reset on the routine before playing it again. 

I don't know why waiting in a s.bind causes things to break though, it shouldn't, and is definitely a bug.
